lib/cc/engine/analyzers/file_thread_pool.rb

```diff
@@ -17,11 +17,8 @@ def run(&block)
 
           @workers = thread_count.times.map do
             Thread.new do
-              begin 
-                while item = queue.pop(true)
-                  yield item
-                end
-              rescue ThreadError
+              while !queue.empty? && item = queue.pop(true)
```

Comment:
What do you think of using `#any?` here? If CC were running on this repo, I think it would recommend wrapping the assignment.

```ruby
while queue.any? && (item = queue.pop(true))
```